e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js

```diff
@@ -1524,3 +1524,61 @@ describe("issue 44668", () => {
     });
   });
 });
+
+describe(
+  "issue 38989",
+  {
+    tags: ["@external"],
+  },
+  () => {
+    beforeEach(() => {
+      restore("postgres-writable");
+      cy.signInAsAdmin();
+    });
+
+    it("should be impossible to join with a table or question which is not in the same database (metabase#38989)", () => {
+      startNewQuestion();
+      entityPickerModal().within(() => {
+        entityPickerModalTab("Tables").click();
+        cy.findByText("Sample Database").click();
+        cy.findByText("Orders").click();
+      });
+
+      cy.button("Visualize").click();
+      cy.button("Save").click();
+      modal().button("Save").click();
+
+      cy.wait(300);
+      modal().button("Not now").click();
+
+      startNewQuestion();
+      entityPickerModal().within(() => {
+        entityPickerModalTab("Tables").click();
+        cy.findByText("Writable Postgres12").click();
+        cy.findByText("Scoreboard Actions").click();
+      });
+
+      join();
+
+      entityPickerModal().within(() => {
+        entityPickerModalTab("Recents").click();
+        cy.findByText("Orders").click();
+      });
```

Comment:
We are selecting a question not a table. This is still a bug, and I filed an issue for it: https://github.com/metabase/metabase/issues/44974

Reply:
So this repro will become invalid once #44974 is fixed (repros should not assume some other bug is there).

Maybe let's create questions programatically and then do the assertion?
It will be a valid repro for existing questions referencing multiple databases.

Reply:
Good suggestion: I have done just that! Also has the benefit of not depending on postgres anymore.